src/AwsMetadata.js

```diff
@@ -10,10 +10,56 @@ export default class AwsMetadata {
 
   constructor(config = {}) {
     this.logger = config.logger || new Logger();
-    this.host = config.host || '169.254.169.254';
+    this.ecs = !!config.ecs;
+    this.host = this.ecs ? '169.254.170.2' : '169.254.169.254';
```

Comment:
Why are these IP addresses hardcoded?

Reply:
It's the instance metadata service.

Reply:
Even Amazon hard-codes these: https://github.com/aws/aws-sdk-js/blob/2b6bcbdec1f274fe931640c1b61ece999aae7a19/lib/metadata_service.js#L25